lib/OpenQA/Utils.pm

```diff
@@ -762,14 +695,13 @@ sub loaded_modules {
 
 # Fallback to loaded_modules if no arguments are given.
 # Accepts namespaces as arguments. If supplied, it will filter by them
-sub loaded_plugins {
-    my $ns = join('|', map { quotemeta } @_);
-    return @_ ? grep { /^$ns/ } loaded_modules() : loaded_modules();
+sub loaded_plugins (@ns) {
+    my $ns = join('|', map { quotemeta } @ns);
+    return @ns ? grep { /^$ns/ } loaded_modules() : loaded_modules();
 }
 
 # Walks a hash keeping keys as a stack
-sub hashwalker {
-    my ($hash, $callback, $keys) = @_;
+sub hashwalker ($hash, $callback, $keys = undef) {
     $keys = [] if !$keys;
```

Comment:
Why not:
```suggestion
sub hashwalker ($hash, $callback, $keys = []) {
```

Reply:
Because that would be an incompatible change, e.g. calls like `hashwalker($hash, $callback, $keys)` would cease to work if `$keys` is `undef`. Of course we *could* change it like you suggest but only after checking that no calls like in my example are made.

Reply:
Which again are not directly related to introducing signatures